test/parallel/test-tls-addca.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+const common = require('../common');
+const fs = require('fs');
+
+if (!common.hasCrypto) {
+  common.skip('missing crypto');
+  return;
+}
+const tls = require('tls');
+
+function filenamePEM(n) {
+  return require('path').join(common.fixturesDir, 'keys', n + '.pem');
+}
+
+function loadPEM(n) {
+  return fs.readFileSync(filenamePEM(n));
+}
+
+var caCert = loadPEM('ca1-cert');
+var contextWithoutCert = tls.createSecureContext({});
+var contextWithCert = tls.createSecureContext({});
+// Adding a CA certificate to contextWithCert should not also add it to
+// contextWithoutCert. This is tested by trying to connect to a server that
+// depends on that CA using contextWithoutCert.
+contextWithCert.context.addCACert(caCert);
+
+var serverOptions = {
+  key: loadPEM('agent1-key'),
+  cert: loadPEM('agent1-cert'),
+};
+var server = tls.createServer(serverOptions, function() { });
+
+var clientOptions = {
+  port: undefined,
+  ca: [caCert],
+  servername: 'agent1',
+  rejectUnauthorized: true,
+};
+
+function startTest() {
+  // This client should fail to connect because it doesn't trust the CA
+  // certificate.
+  clientOptions.secureContext = contextWithoutCert;
+  clientOptions.port = server.address().port;
+  var client = tls.connect(clientOptions, common.fail);
+  client.on('error', function() {
```

Comment:
Let's wrap function with `common.mustCall`, and I guess we may use arrow function here:

```client.on('error', common.mustCall(() => {```